syntax = "proto3";

package ticketing;

import "google/protobuf/timestamp.proto";

service GrpcTicketingService {
  rpc CheckReservation(CheckReservationRequest) returns (CheckReservationResponse);

  rpc GetTicketPrice(GetTicketPriceRequest) returns (GetTicketPriceResponse);
}

message GetTicketPriceRequest {
    string ticket_type_id = 1;
}

message GetTicketPriceResponse {
    optional string error = 1;
    string amount = 2;
}

message CheckReservationRequest {
    string reservation_id = 1;
}

message Reservation {
    string id = 1;
    string user_id = 2;
    google.protobuf.Timestamp created_at = 3;
    google.protobuf.Timestamp expires_at = 4;
    string status = 5;
}

message CheckReservationResponse {
    bool exists = 1;
    bool valid = 2;
    Reservation reservation = 3;
    string error = 4;
}